### DISCUSSION TITLE
AWS Marketplace version is outdated

### DISCUSSION THREAD

**Discussion by Asterovim** [langgenius/dify#3715] [Help]
https://aws.amazon.com/marketplace/pp/prodview-t22mebxzwjhu6
I'm noticing the current version of the product on the AWS Marketplace is 0.5.8. Could you please clarify why this version seems outdated?

Additionally, if I were to subscribe to this product, how would the update process work?

**Comment by crazywoola** [comment]
There is a process for listing on AWS, and we will update to the latest version as soon as possible. At the time, 0.5.6 was also a milestone release and our first submission to the Marketplace.

**Reply by mamift** [reply]
Can you at least inform us when the AWS AMI will be updated to the 0.15.x [release version](https://github.com/langgenius/dify/discussions/13326)? As of Feb 2025, the AMI's latest update is 0.6.13 which was from July 2024: https://aws.amazon.com/marketplace/pp/prodview-t22mebxzwjhu6

It would also be good to if you could post on your [docs](https://docs.dify.ai/getting-started/dify-premium-on-aws) how often you publish updated AMI images. 

Also, the main benefit of an AMI is to make the cloud operational burden easier while also maintaining control over the data, which is a common requirement for many businesses, especially ones that need to geofence client data. If we were to deploy a new EC2 instance right now, we'd then need to immediately run a bunch of manual [upgrade commands](https://docs.dify.ai/getting-started/dify-premium-on-aws#upgrading) to get the latest version of Dify, which kind of negates the benefits of an AMI, where many other vendors handle updates for the customer through OS-level updating. 

Also, I think it would also be highly beneficial to implement some sort of auto-download of updates via the AMI. From a customer perspective, it would be great to just deploy an AMI, and have it auto-download updates, then from the admin interface, be able to schedule an automatic restart to update. Currently, we have to login manually to the EC2 instance and run a bunch of manual update commands. 

In the US East (N. Virginia), subscribing to Dify Premium AMI is $0.3 per hour, so that's $7.2, for 365 days, that's $2628 - that doesn't include hardware costs. As it is, there's not much more value-add aside from the promise of better support.

**Reply by patricia-cmc** [reply]
I agree, I'm struggling to upgrade this old version.
The Documentation doesn't contain enough information of the upgrade procedure.
I'm struggling to upgrade following the procedure in the https://docs.dify.ai/getting-started/dify-premium-on-aws which by the way it doesn't work...
I'm getting permission errors. Is incredible frustrating and I can't believe this is a product released through AWS Marketplace...
What exactly is "Premium" about and old version of Dify where we have to do all the work for manual updates?

**Reply by lethao911** [reply]
Have you bought it yet?

**Reply by mamift** [reply]
I did buy it (had an EC2 instance running), up until last week; but as it's outdated by 7 months and cannot be updated to the latest version as per the [instructions here](https://docs.dify.ai/getting-started/dify-premium-on-aws) I terminated the instance as it's literally easier to self-host the community version.

**Comment by mamift** [comment]
Also, when will the AWS AMI be updated to version 1.0? https://aws.amazon.com/marketplace/pp/prodview-t22mebxzwjhu6

Now that v1 has launched, it would be nice to update the AMI to this version. It's still at 0.6.13 as of writing. I would be willing to deploy an EC2 instance using a v1.0 AMI, as at least we can get the last 6 months worth of changes.

**Comment by lethao911** [comment]
Anyone from the Dify team here? I don't see any response?

**Comment by mamift** [comment]
OK so it seems the v1.4.4 AMI has been published on the AWS marketplace: https://aws.amazon.com/marketplace/pp/prodview-t22mebxzwjhu6

I got an email about it from Amazon just yesterday.

However it would be nice to know about the future AWS roadmap and how often the AMI image is going to be updated; will it stay in sync with Github releases? Or will it lag by n-releases or versions?
